src/Listeners/InspectorListener.php

```diff
@@ -169,4 +277,30 @@ protected function notifyUnexpectedError(Throwable $throwable): void
     {
         $this->inspector->reportException($throwable, false);
     }
+
+    /**
+     * Workaround method, should be removed after
+     * @link https://github.com/inspector-apm/inspector-php/issues/9
+     */
+    private function startSegment(string $label): void
+    {
+        $segment = $this->inspector->startSegment(self::SEGMENT_TYPE_PROCESS, $label);
+
+        $this->segments[$label] = $segment;
+    }
+
+    /**
+     * Workaround method, should be removed after
+     * @link https://github.com/inspector-apm/inspector-php/issues/9
+     */
+    private function endSegment(string $label): void
+    {
+        if (!isset($this->segments[$label])) {
```

Comment:
I prefer to use "array_as_key" php native function to perform this check. It's a little more verbose also for readability.